autoprecompiles/src/trace_handler.rs

```diff
@@ -108,3 +119,12 @@ impl<F> Trace<F> {
         Self { values, width }
     }
 }
+
+pub fn generate_trace<'a, A: Adapter>(
+    air_id_to_dummy_trace: &'a HashMap<A::AirId, Trace<A::Field>>,
+    instruction_handler: &'a A::InstructionHandler,
+    apc_call_count: usize,
+    apc: &Apc<A::Field, A::Instruction>,
+) -> TraceHandlerData<'a, A::Field> {
+    TraceHandler::<A>::new(air_id_to_dummy_trace, instruction_handler, apc_call_count).data(apc)
```

Comment:
In a different PR, we could inline these calls and remove the trace handler struct altogether